Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -478,7 +490,14 @@ static gboolean webkitMediaCommonEncryptionDecryptSinkEventHandler(GstBaseTransf
             result = TRUE;
             LockHolder locker(priv->m_mutex);
             RELEASE_ASSERT(webkitMediaCommonEncryptionDecryptIsCDMInstanceAvailable(self));
-            priv->m_keyReceived = klass->attemptToDecryptWithLocalInstance(self, priv->m_initDatas.get(WebCore::GStreamerEMEUtilities::keySystemToUuid(priv->m_cdmInstance->keySystem())));
+            auto initData = priv->m_initDatas.get(WebCore::GStreamerEMEUtilities::keySystemToUuid(priv->m_cdmInstance->keySystem()));
+            auto keyId = !initData.isEmpty() ? priv->m_keyIds.get(initData) : nullptr;
+            GST_DEBUG_OBJECT(self, "attempting to decrypt with local instance %p, key id %p", priv->m_cdmInstance.get(), keyId);
+            if (keyId) {
+                GST_MEMDUMP_OBJECT(self, "key id", reinterpret_cast<const guint8*>(keyId->data()), keyId->size());
+                Ref<WebCore::SharedBuffer> protectedKeyId(*keyId);
+                priv->m_keyReceived = klass->attemptToDecryptWithLocalInstance(self, WTFMove(protectedKeyId));
```

Comment:
`auto keyId`  becomes RefPtr<SharedBuffer>, not even a reference because you could be constructing it also with `nullptr`. You don't need to "protect" it and move it, you should be able to just move it.

Reply:
TBH this is something which got me. auto keyId is SharedBuffer* due to traits for Ref<> in HasMap<> and without protecting it got released after the scope.

Reply:
Sigh... traits... They always bit me and beat me. Anyway. I think I comment below that we don't need a &&, but rather a `const SharedBuffer&`, so I guess the comment still holds somehow.

Reply:
Yup